### DISCUSSION TITLE
Bug (?) GitHub Actions run on Pull Request from main branch

### DISCUSSION THREAD

**Discussion by ghost** [community/community#26876] [Actions]
<p>In a GitHub Actions file that is set to run like this</p>
<pre><code class="lang-auto">on:
  push:
    branches:
      - branch_name
</code></pre>
<p>The action will run when you create a pull request from branch_name to another branch.<br>
This issue could be very bad when someone has a fork of your repository and creates a pull request from branch_name to their own fork, which would trigger the Action despite it not being a push to branch_name on origin.</p>
<p>I hope this is not a duplicate, I couldn’t find any reports of this issue when attempting to search for it here and on Google.</p>

**Comment by BrightRan** [comment, marked as answer]
<p><a class="mention" href="/natsumihb">@NatsumiHB</a> ,</p>
<p>According to the configurations, this workflow only can be triggered when pushing new commits to the <strong>branch_name</strong>.<br>
When you open a PR to merge commits from <strong>branch_name</strong> into another branch, the <strong>branch_name</strong> is the head branch (source branch) of the PR. There is a point you need to understand:</p>
<ul>
<li>All the jobs in the workflow run that triggered by the latest commit push on the head branch will be listed as the checks of the PR.<br>
<strong>Note:</strong> it just fetches jobs from the existing workflow run that triggered by the <strong>push</strong> event on the head branch, it does not trigger new workflow run on the head branch.</li>
</ul>
<p>If the PR indeed triggers a new workflow run on the head branch, please share your repository with us so that we can check more detailed configurations of your workflows to analyze the root cause.</p>

**Comment by ghost** [comment]
<p>That makes total sense to me and I believe that’s what happened. I probably thought that it ran the workflow again because it didn’t finish yet but in reality it just showed the progress.</p>

**Comment by BrightRan** [comment]
<p><a class="mention" href="/natsumihb">@NatsumiHB</a> ,</p>
<p>So, you just have a misunderstanding for the checks listed on the PR, and my above explanation can answer your doubt, right?</p>
<p>In addition, generally the following things will be listed as the checks of the PR:</p>
<ol>
<li>
<p>The jobs in the workflows triggered by the events related to pull request.</p>
<ul>
<li><a href="https://docs.github.com/en/actions/reference/events-that-trigger-workflows#pull_request" rel="noopener nofollow ugc"><strong>pull_request</strong></a></li>
<li><a href="https://docs.github.com/en/actions/reference/events-that-trigger-workflows#pull_request_review" rel="noopener nofollow ugc"><strong>pull_request_review</strong></a></li>
<li><a href="https://docs.github.com/en/actions/reference/events-that-trigger-workflows#pull_request_review_comment" rel="noopener nofollow ugc"><strong>pull_request_review_comment</strong></a></li>
<li><a href="https://docs.github.com/en/actions/reference/events-that-trigger-workflows#pull_request_target" rel="noopener nofollow ugc"><strong>pull_request_target</strong></a></li>
</ul>
</li>
<li>
<p>As mentioned above, the jobs in the workflows triggered by <a href="https://docs.github.com/en/actions/reference/events-that-trigger-workflows#push" rel="noopener nofollow ugc">push</a> event on the head branch.</p>
</li>
<li>
<p>Some check runs created from integrated GitHub Apps for the PR. For example, an Azure Pipeline triggered by the PR.</p>
</li>
</ol>
<p>If my above answer is helpful to you, maybe you can mark it as the solution of this topic, so that other people who have the similar questions can notice it when they are looking for an answer.</p>
<p>If you have any other questions about this topic, feel free to tell us. Thanks.</p>

**Comment by ghost** [comment]
<p>Yep, that’s exactly what happened! I marked your reply as the answer.</p>
<p>And thank you for your very good explanation!</p>

**Comment by BrightRan** [comment]
<p>You’re welcome, <a class="mention" href="/natsumihb">@NatsumiHB</a> .</p>
